### PR TITLE
Adding check to verify secret passed in is for webhooks

### PR DESCRIPTION
**Pull request by ctrudeau-stripe**
When using webhooks you need to pass in the `whsec_` found in the dashboard webhooks section. If you pass in the secret key not related to webhooks it leads to a confusing error. This adds a check to verify it's the webhook secret being passed in. 

r? @richardm-stripe

### CODE REVIEW

**Review by richardm-stripe — APPROVED** [review]

File: lib/Webhooks.js [hunk 1]
```diff
@@ -104,6 +104,13 @@ const signature = {
       });
     }
 
+    if (secret.substring(0, 6) != 'whsec_') {
```
Comment (paulasjes-stripe) [inline lib/Webhooks.js:107]: Can simplify with `!secret.startsWith('whsec_')`

File: lib/Webhooks.js [hunk 2]
```diff
@@ -104,6 +104,13 @@ const signature = {
...
+      throw new StripeSignatureVerificationError({
+        message:
+          'Webhook secret key passed in does not start with whsec_. Webhook secrets can be found here: https://dashboard.stripe.com/webhooks.',
```
Comment (paulasjes-stripe) [inline lib/Webhooks.js:110]: Suggest rewording this for clarity:

`Wrong signing secret provided. Please ensure you are passing in a valid webhook signing secret that starts with whsec_. Webhook secrets can be found at: https://dashboard.stripe.com/webhooks.`

**Comment by remi-stripe** [comment]
I'm quite ambivalent to merge this as is because the secret is opaque and if we ever change the prefix, like we did for API keys multiple times, then it will lead to confusing error messages for developers trying to debug their code.

**Comment by paulasjes-stripe** [comment]
@remi How would you feel about instead of checking for `whsec_` we check that length of the key passed in is larger than 0 and that the key doesn't begin with `sk_` or `pk_`? Seems unlikely that we'll ever use those prefixes for a webhook signing secret.

Reason for this PR is that a user got confused and passed in their secret key rather than a webhook signing secret.

**Comment by remi-stripe** [comment]
Maybe, but there are other keys prefix like rk_ and ek_. Overall, I would personally punt on this kind of helper methods in client libraries.

File: lib/Webhooks.js (hunk 1 above)
Comment (ctrudeau-stripe) [reply lib/Webhooks.js:107]: I went with a solution I felt was the most backwards compatible. I'm not sure which version we target for node. If we are up to date and can do startsWith and that won't break anyone.

**Comment by ctrudeau-stripe** [comment]
Punting to find a different solution.
